subprojects/testing-base/src/integTest/groovy/org/gradle/testing/TestExecutionBuildOperationsIntegrationTest.groovy

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing
+
+import org.gradle.api.internal.tasks.testing.operations.ExecuteTestBuildOperationType
+import org.gradle.api.internal.tasks.testing.operations.TestOutputBuildOperationType
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+import org.gradle.integtests.fixtures.TestResources
+import org.gradle.internal.operations.trace.BuildOperationRecord
+import org.junit.Rule
+
+import static org.gradle.util.TextUtil.normaliseLineSeparators
+
+class TestExecutionBuildOperationsIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
We also have XCTests -- do we need to a separate test for build operations for that as well?

Maybe we could move the first two "emits..." tests into `AbstractTestFrameworkIntegrationTest`?

Reply:
@big-guy I'll address this in a follow on on master. In general we also want XCTests exposed by that.